src/authorization.js

```diff
@@ -258,6 +268,12 @@ class Authorization {
     return this.accessModes[ Authorization.acl.CONTROL ]
   }
 
+  clone () {
+    let auth = new Authorization()
+    Object.assign(auth, JSON.parse(JSON.stringify(this)))
```

Comment:
I see. I scanned over that JSON.parse/JSON.stringify expression. I have no idea regarding performance not having profiled either approach.